galaxy.py

```diff
@@ -69,97 +55,84 @@ def get_galaxy_connection( use_objects=DEFAULT_USE_OBJECTS ):
         through. This will succeed where the previous connection fails under
         the conditions of REMOTE_USER and galaxy running under uWSGI.
     """
-    conf = _get_conf()
-    key = conf['api_key']
+    history_id = history_id or os.environ['HISTORY_ID']
+    key = os.environ['API_KEY']
 
     ### Customised/Raw galaxy_url ###
     galaxy_ip = _get_ip()
     # Substitute $DOCKER_HOST with real IP
-    url = Template(conf['galaxy_url']).safe_substitute({'DOCKER_HOST': galaxy_ip})
-    gi = _test_url(url, key, history_id, use_objects=use_objects)
+    url = Template(os.environ['GALAXY_URL']).safe_substitute({'DOCKER_HOST': galaxy_ip})
+    gi = _test_url(url, key, history_id)
     if gi is not None:
         return gi
 
     ### Failover, fully auto-detected URL ###
     # Remove trailing slashes
-    app_path = conf['galaxy_url'].rstrip('/')
+    app_path = os.environ['GALAXY_URL'].rstrip('/')
     # Remove protocol+host:port if included
     app_path = ''.join(app_path.split('/')[3:])
 
-    if 'galaxy_paster_port' not in conf:
+    if 'GALAXY_WEB_PORT' not in os.environ:
         # We've failed to detect a port in the config we were given by
         # galaxy, so we won't be able to construct a valid URL
         raise Exception("No port")
     else:
         # We should be able to find a port to connect to galaxy on via this
         # conf var: galaxy_paster_port
-        galaxy_port = conf['galaxy_paster_port']
+        galaxy_port = os.environ['GALAXY_WEB_PORT']
 
     built_galaxy_url = 'http://%s:%s/%s' %  (galaxy_ip.strip(), galaxy_port, app_path.strip())
     url = built_galaxy_url.rstrip('/')
 
-    gi = _test_url(url, key, history_id, use_objects=use_objects)
+    gi = _test_url(url, key, history_id)
     if gi is not None:
         return gi
 
     ### Fail ###
-    raise Exception("Could not connect to a galaxy instance. Please contact your SysAdmin for help with this error")
-
-
-def _get_history_id():
-    """
-    Extract the history ID from the config file.
-    """
-    conf = _get_conf()
-    return conf['history_id']
-
-
-def put(filename, file_type='auto', history_id=None, use_objects=DEFAULT_USE_OBJECTS):
+    msg = "Could not connect to a galaxy instance. Please contact your SysAdmin for help with this error"
+    if '127.0.0.1' in os.environ['GALAXY_URL']:
+        msg += (
+            "\nYou seem to be running galaxy on localhost. "
```

Comment:
What host is galaxy bound to? Is it bound to 127.0.0.1 or 0.0.0.0/your-ip-address?

I access my local dev galaxy instance at this address as well, however, it's bound to 0.0.0.0, as otherwise requests from other ip addresses are rejected, including those from docker.